File: turboc/src/io/wrapper/Vector3DPrinter.cpp
#include "io/wrapper/Vector3DPrinter.h"

void
Vector3DPrinter::print(const Vector3D &vector, PrintStream *stream) {
    if ( stream == NULL ) {
        return;
    }
    stream->printf("%g %g %g", vector.x, vector.y, vector.z);
}

File: typescript/src/main/vsdk/toolkit/io/wrapper/Vector3DPrinter.ts
import { PrintStream } from "../../../../java/io/PrintStream";
import { Vector3D } from "../../common/linealAlgebra/Vector3D";

export class Vector3DPrinter {
  private static toGeneralNumber(value: number): string {
    return Number.isFinite(value) ? value.toString() : `${value}`;
  }

  public static print(vector: Vector3D, stream: PrintStream | null): void {
    if (stream === null) {
      return;
    }
    stream.printf(
      "%s %s %s",
      Vector3DPrinter.toGeneralNumber(vector.x),
      Vector3DPrinter.toGeneralNumber(vector.y),
      Vector3DPrinter.toGeneralNumber(vector.z)
    );
  }
}
